### ISSUE TITLE
`task: distill` accepts rows with no loss target, which SFT refuses: the step loss is NaN, the logged training loss becomes 0.0 or doubles, and `eval_loss` is NaN

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1242]
## What happens

A chat row whose prompt is longer than `data.max_length` loses its whole assistant reply to truncation, leaving no label that is not -100. SFT refuses such a row at setup and names it. The same data on `task: distill` is accepted, and every microbatch made of that row has a NaN loss.

Reproduced on CPU with the real `DistillTrainerWrapper` (default token-level `forward_kl`). The student and teacher are tiny random-init Llamas built from a config and sharing one word-level tokenizer (no download). The config uses `data.max_length: 64`, `chat_template: chatml`, `batch_size: 1` and LoRA r=4. `SHORT` is a normal row; `LONG` has a 96-word user message. A spy around `compute_loss` recorded each call:

```
[ga1_long_row] GA=1 | supervised labels per train row after truncation: [10, 0]
[ga1_long_row] compute_loss calls (supervised labels, loss): [(10, 2.1389989852905273), (0, nan)]
[ga1_long_row] logged (loss, grad_norm): [(2.1389989852905273, 0.7034521102905273), (0.0, 0.0)]

[ga1_only_long_row] GA=1 | supervised labels per train row after truncation: [0, 0]
[ga1_only_long_row] compute_loss calls (supervised labels, loss): [(0, nan), (0, nan)]
[ga1_only_long_row] logged (loss, grad_norm): [(0.0, 0.0), (0.0, 0.0)]

[default_ga] GA=4 | supervised labels per train row after truncation: [10, 0]
[default_ga] compute_loss calls (supervised labels, loss): [(10, 2.1389989852905273), (0, nan)]
[default_ga] logged (loss, grad_norm): [(4.277997970581055, 0.7034521102905273)]

[eval_split_long_row] evaluate(): {'eval_loss': nan, 'eval_runtime': 0.0116, 'eval_samples_per_second': 86.085, 'eval_steps_per_second': 86.085, 'epoch': 1.0}
```

So:

- A step whose only microbatch has no target is **logged as loss 0.0**. A dataset in which every row is truncated like this trains nothing and reports a perfect loss.
- With the default `gradient_accumulation_steps: 4`, the NaN microbatch shares a window with a real one, and the **logged loss is doubled**: 4.278 against a true window loss of 2.139.
- An eval split containing such a row gives **`eval_loss: nan`**.

The same rows on `task: sft`:

```
SFT setup refused: train row 2: no causal-loss target remains after tokenization/truncation at data.max_length=64; the assistant response is absent or fully truncated. Increase data.max_length or shorten the prompt.
```

**What did not reproduce: NaN gradients and a NaN adapter.** On CPU the gradients of a mean cross-entropy over zero targets are exactly 0, and in all four runs above every LoRA tensor was finite after training:

```
mean CE over zero targets = nan | grad finite: True | grad abs max: 0.0
token-weighted (loss*0/1) = nan | grad finite: True
[ga1_long_row] every LoRA tensor finite after training: True (8 tensors)
```

#532 saw an all-NaN adapter from the SFT version of this bug on MPS with bf16 layer streaming. Whether distill can reach that on MPS or CUDA was not tested here, and this issue does not claim it.

## Why

`src/soup_cli/trainer/distill.py:674-689` maps the train and val rows with `build_format_row` directly. SFT wraps the same mapping in `ensure_causal_loss_target` (`trainer/sft.py:137-166`, using `data/loss_mask.py:233`), which raises `NoCausalLossTargetError` with the row number. In `compute_loss`, `distill.py:807-815` takes a mean `cross_entropy(..., ignore_index=-100)` over zero targets, which is `0/0 = NaN`, and `_token_weighted_accumulation` (`:781-805`) multiplies it by 0, which is still NaN. The Hugging Face Trainer's `logging_nan_inf_filter` (on by default; `transformers/trainer.py:1845-1850`) replaces a NaN step loss with `tr_loss / (1 + steps since the last log)`. That gives 0.0 when nothing real was accumulated and a doubled value when something was. Evaluation has no such filter. The non-finite tracker added in #887 watches only the distillation term, and that term returns a finite 0 for an empty mask (`distill.py:137-139`), so it stays silent.

## Fix path

Apply the SFT guard to distill: map the distill train and val rows through the same wrapper as `_map_text_sft_rows` (or call `ensure_causal_loss_target` in the distill map), so the row is refused at setup with the same message as SFT, naming the row and `data.max_length`. As defence in depth, the CE term in `compute_loss` could return 0 rather than NaN when a microbatch has no targets. The refusal is the user-facing fix, though, because a silently skipped row is still a data problem the user should hear about.

## Acceptance

- [ ] A CPU test: `DistillTrainerWrapper.setup({"train": [SHORT, LONG]})` with `data.max_length: 64` raises `ValueError` whose message contains `train row 2` and `data.max_length=64`. Currently it is accepted and `compute_loss` returns NaN for that row.
- [ ] The same for the val split (`val row N`).
- [ ] Variants: the default token-level path (`forward_kl`, `reverse_kl`, `js`); `uld_strategy`; `minillm_enabled`; `distill_mode: sequence`, where the teacher can generate an empty completion and leave no target; `data.train_on_messages_with_train_field` with every message marked `train: false`; `gradient_accumulation_steps` 1 and 4.
- [ ] A row with at least one surviving target is still accepted (control).
